src/main/java/ua/rent/masters/easystay/service/impl/BookingServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package ua.rent.masters.easystay.service.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ua.rent.masters.easystay.dto.request.BookingRequestDto;
+import ua.rent.masters.easystay.dto.request.BookingRequestUpdateDto;
+import ua.rent.masters.easystay.dto.response.BookingResponseDto;
+import ua.rent.masters.easystay.dto.response.BookingResponseUpdatedDto;
+import ua.rent.masters.easystay.mapper.BookingMapper;
+import ua.rent.masters.easystay.model.Booking;
+import ua.rent.masters.easystay.model.BookingStatus;
+import ua.rent.masters.easystay.repository.BookingRepository;
+
+@Service
+@RequiredArgsConstructor
+public class BookingServiceImpl implements BookingService {
+
+    private final BookingMapper bookingMapper;
+    private final BookingRepository bookingRepository;
+
+    @Override
+    public BookingResponseDto create(BookingRequestDto requestDto) {
+        validateAccommodationId(requestDto.accommodationId());
+        Booking booking = bookingMapper.toEntity(requestDto);
+        booking.setStatus(BookingStatus.PENDING);
+
+        Booking savedBooking = bookingRepository.save(booking);
+        return bookingMapper.toDto(savedBooking);
+    }
+
+    @Override
+    public BookingResponseDto getBookingByUserIdByStatus(Long userId, BookingStatus bookingStatus) {
+        return null;
+    }
+
+    @Override
+    public List<BookingResponseDto> getAll() {
+        return bookingRepository.findAll().stream()
+                .map(bookingMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookingResponseDto getById(Long bookingId) {
+        Booking booking = getBookingByIdOrThrowException(bookingId);
+        return bookingMapper.toDto(booking);
+    }
+
+    @Override
+    public BookingResponseUpdatedDto updateById(Long bookingId,
+                                                BookingRequestUpdateDto requestUpdateDto) {
+        Booking booking = getBookingByIdOrThrowException(bookingId);
+        booking.setCheckInDate(requestUpdateDto.checkInDate());
+        booking.setCheckOutDate(requestUpdateDto.checkOutDate());
+        booking.setAccommodationId(requestUpdateDto.accommodationId());
+        booking.setStatus(BookingStatus.PENDING);
```

Comment:
Why you hardcode PENDING?

Reply:
Try to use 
@BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
    @Mapping(target = "id", ignore = true)
    void updateBookingFromDto(BookingRequestUpdateDto dto, @MappingTarget Booking entity);